src/_common/settings/options.service.ts

```diff
@@ -0,0 +1,74 @@
+import { assertNever } from '../../utils/utils';
+const STORAGE_PREFIX = 'options.';
+
+type OptionsType = 'community-eject' | 'community-move-post' | 'community-user-block';
+
+export function getOptions(type: OptionsType, suffix = '') {
+	const id = type + (suffix ? '-' + suffix : '');
+	switch (type) {
+		case 'community-eject':
+		case 'community-move-post':
+			return new Options<string>(id, true, 10);
+		case 'community-user-block':
+			return new Options<string>(id, true, 20);
+		default:
+			assertNever(type);
+	}
+}
+
+export class Options<T> {
+	private _storageKey: string;
+	private _checkDuplicate: boolean;
+	private _maxItems: number | null;
+
+	constructor(id: string, checkDuplicate: boolean, maxItems: number | null) {
```

Comment:
lets get rid of checkDuplicate. I don't see where we'd even want to store duplicates in the first place